pkg/registry/cluster/strategy.go

```diff
@@ -76,8 +77,8 @@ func (Strategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
 
 // Validate returns an ErrorList with validation errors or nil.
 func (Strategy) Validate(ctx context.Context, obj runtime.Object) field.ErrorList {
-	// TODO: add validation for Cluster
-	return field.ErrorList{}
+	cluster := obj.(*clusterapis.Cluster)
```

Comment:
This obj will always be `clusterapis.Cluster`? not `clusterv1alpha1.Cluster`? I'm not sure about it.

Reply:
@RainbowMango Yes, the versioned object is always converted into the internal object in the chain of apiserver hander.

https://github.com/karmada-io/karmada/blob/master/vendor/k8s.io/apiserver/pkg/endpoints/handlers/create.go#L120

```go
        decoder := scope.Serializer.DecoderToVersion(s.Serializer, scope.HubGroupVersion)
        original := r.New()
        // defaultGVK = {Group: cluster, Version: v1alpha1, Kind: Cluster}
        // originalGVK = {Group: cluster, Version: __internal, Kind: Cluster}
	obj, gvk, err := decoder.Decode(body, &defaultGVK, original)
        r.Create(ctx, name,obj, est.AdmissionToValidateObjectFunc(admit, admissionAttributes, scope), options)
```